less_slow.cpp

```diff
@@ -4984,12 +5056,40 @@ BENCHMARK(errors_with_status)->ComputeStatistics("max", get_max_value)->MinTime(
 
 using std::string_view_literals::operator""sv;
 
+template <typename logger_type_>
+static void logging(bm::State &state) {
+    struct {
+        int code;
+        std::string_view message;
+    } errors[3] = {
+        {1, "Operation not permitted"sv},
+        {12, "Cannot allocate memory"sv},
+        {113, "No route to host"sv},
+    };
+    char buffer[1024];
+    logger_type_ logger;
+    std::size_t iteration_index = 0;
+    std::size_t bytes_logged = 0;
+    for (auto _ : state) {
+        bytes_logged += logger(              //
+            buffer, sizeof(buffer),          //
+            std::source_location::current(), //
+            errors[iteration_index % 3].code, errors[iteration_index % 3].message);
+        iteration_index++;
+    }
+    state.SetBytesProcessed(bytes_logged);
+}
+
 struct log_printf_t {
     std::size_t operator()(                    //
         char *buffer, std::size_t buffer_size, //
         std::source_location const &location, int code, std::string_view message) const noexcept {
-
+#if defined(_MSC_VER)
+        // On MSVC, high_resolution_clock is steady_clock, which cannot have to_time_t applied to it 
+        auto now = std::chrono::system_clock::now();
+#else
         auto now = std::chrono::high_resolution_clock::now();
+#endif
```

Comment:
This is probably a good place to add some notes on how `std::chrono` ambiguously wraps different system-level APIs without providing many guarantees 🤔

Reply:
True, I'll make some notes about that, and about how high_resolution_clock is implementation-defined.